### PR TITLE
Parents track not available

### PR DESCRIPTION
**Pull request by adityaekawade**
This PR fixes the issue https://github.com/iobio/gene.iobio.vue/issues/653 and #609 

To test: 
Launch gene.iobio from quarantest. 
(You will get a warning: Unable to access OMIM, click OK)
Select Load your data
Select trio and GRcH38
Use the following URL's
https://iobio.s3.amazonaws.com/samples/bam/NA12878.GRCh38.bam
https://iobio.s3.amazonaws.com/samples/bam/NA12891.GRCh38.bam
https://iobio.s3.amazonaws.com/samples/bam/NA12892.GRCh38.bam

https://iobio.s3.amazonaws.com/samples/vcf/platinum-exome.GRCh38.vcf.gz

Add gene F5
Check if all tracks are visible 
Screen Shot 2020-11-19 at 2 57 18 PM

 Refresh the link, so gene.iobio is now launched with URL. Check if all tracks are available. 

Also the hyperlink on the title gene.iobio should be removed.

### CODE REVIEW

**Comment by anderspitman** [comment]
quarantest build successful. Demo link at
http://904d17e1.quarantest.iobio.io

File: client/app/components/pages/GeneHome.vue [hunk 1]
```diff
@@ -2805,6 +2806,12 @@ export default {
           }
         }
 
+        let isTrio = self.paramRelationships.every(sample => sample);
+        if (isTrio) {
+          self.isMother = true;
+          self.isFather = true;
+        }
```
Comment (adityaekawade) [inline client/app/components/pages/GeneHome.vue:2813]: If the gene is launched from the URL this checks if the `paramRelationships` array has the value "mother" and "father" in the array or if it is undefined. If it is true, it sets `isMother` and `isFather` as true. 

`paramRelationships` is set from the route parameters and is an array with 3 indexes. If it is a singleton, the last 2 indexes are set as undefined.

File: client/app/components/partials/FilesDialog.vue [hunk 2]
```diff
@@ -513,8 +513,10 @@ export default {
     },
     onModeChanged: function() {
       if (this.mode == 'trio' && this.cohortModel.getCanonicalModels().length < 3 ) {
+        this.$emit('isTrio', true);
```
Comment (adityaekawade) [inline client/app/components/partials/FilesDialog.vue:516]: If the `trio` option is selected on the files dialog when uploading the data, it sets the flag `isTrio` as true and is passed across the components.

**Comment by anderspitman** [comment]
quarantest build successful. Demo link at
http://bc691dc7.quarantest.iobio.io

**Comment by stefinfection** [comment]
Something still a little bit funky with the parent tracks - father track not showing up on refresh. To reproduce:

1. Do your instructions for loading that trio and choose gene F5.
2. Before refresh step, select gene APC.
3. Then refresh, and select Mother and Father tracks.
4. Father track is not displaying upon selection (just the label)

Screen Shot 2020-11-20 at 11 37 17 AM

**Comment by adityaekawade** [comment]
Good find. It looks like another bug..

**Comment by adityaekawade** [comment]
Hi @stefinfection It looks like actually there are no variants found for the "father" track for gene `APC` and hence it is not displaying. 
To test: Enter the data, select the gene "APC" and load. You can see the father label in optional track but the track is not displayed. 
Screen Shot 2020-11-20 at 1 59 41 PM

Now before the refresh, add the gene `F5` or something else like `PRX`.. 
Refresh, both "Father" and "Mother" tracks should be displayed
Screen Shot 2020-11-20 at 1 57 27 PM
 

For `PRX` gene after refresh:
Screen Shot 2020-11-20 at 2 00 36 PM
URL: http://bc691dc7.quarantest.iobio.io/?gene=PRX&genes=APC,F5,PRX&species=Human&build=GRCh38&affectedSibs=&unaffectedSibs=&rel0=proband&vcf0=https%3A%2F%2Fiobio.s3.amazonaws.com%2Fsamples%2Fvcf%2Fplatinum-exome.GRCh38.vcf.gz&tbi0=&bam0=https%3A%2F%2Fiobio.s3.amazonaws.com%2Fsamples%2Fbam%2FNA12878.GRCh38.bam&bai0=&sample0=NA12878&affectedStatus0=affected&rel1=mother&vcf1=https%3A%2F%2Fiobio.s3.amazonaws.com%2Fsamples%2Fvcf%2Fplatinum-exome.GRCh38.vcf.gz&tbi1=&bam1=https%3A%2F%2Fiobio.s3.amazonaws.com%2Fsamples%2Fbam%2FNA12891.GRCh38.bam&bai1=&sample1=NA12891&affectedStatus1=unaffected&rel2=father&vcf2=https%3A%2F%2Fiobio.s3.amazonaws.com%2Fsamples%2Fvcf%2Fplatinum-exome.GRCh38.vcf.gz&tbi2=&bam2=https%3A%2F%2Fiobio.s3.amazonaws.com%2Fsamples%2Fbam%2FNA12892.GRCh38.bam&bai2=&sample2=NA12892&affectedStatus2=unaffected

**Comment by stefinfection** [comment]
Oh I see - but the coverage tracks should still be displayed for mother and father, right?

**Comment by adityaekawade** [comment]
I am not entirely sure if it was changed or updated recently, but I think the coverage tracks for Mother and Father are shown when a Gene is clicked from the "Genes" tab..  We can create a separate issue for it.

**Comment by stefinfection** [comment]
Ok cool - everything else in this PR looks good to my eyes!

**Comment by AlistairNWard** [comment]
Couple of things:
1. Is it clear to a user that if the track isn't showing, there are no variants? I'm personally an advocate of being explicit with the user. Nothing showing could mean there are no variants, or there was a glitch and they didn't load. If there are no variants, could we explain state that is the case?

2. I followed the link above, and the gene model now appears in the top Gene card. On prod gene, the gene model is only included in the variant cards for each individual. This should be removed from the Gene card.

**Pull request merged**